.github/workflows/test.yml

```diff
@@ -7,67 +7,164 @@ on:
   pull_request:
 
 # Current configuration:
-# - django 2.2, python 3.6, wagtail 2.5, sqlite
 # - django 2.2, python 3.7, wagtail 2.7, postgres
 # - django 3.0, python 3.8, wagtail 2.11, postgres
 # - django 3.1, python 3.8, wagtail 2.12, sqlite
 # - django 3.2, python 3.9, wagtail 2.13, postgres
-# - django 3.2, python 3.9, wagtail main, sqlite (allow failures)
+# - django 3.2, python 3.9, wagtail 2.14, postgres
+# - django 3.2, python 3.9, wagtail 2.15, sqlite
+# - django 3.2, python 3.10, wagtail 2.16, sqlite
+# - django 3.2, python 3.10, wagtail 3.0, postgres
+# - django 3.2, python 3.10, wagtail 4.0, postgres
+# - django 3.2, python 3.10, wagtail main, sqlite (allow failures)
 jobs:
   test:
     runs-on: ubuntu-latest
     continue-on-error: ${{ matrix.experimental }}
     strategy:
       matrix:
         include:
-          - python: 3.6
-            django: "Django>=2.2,<2.3"
-            wagtail: "wagtail>=2.5,<2.6"
-            database: "sqlite3"
+          - python: 3.7
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.1,<4.2"
+            database: "postgresql"
             experimental: false
           - python: 3.7
-            django: "Django>=2.2,<2.3"
-            wagtail: "wagtail>=2.7,<2.8"
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.2,<5.0"
```

Comment:
Suggestion: Again, I was wondering if we should limit to <4.3 instead of <5.0? What do you think?